src/LanguageServer/Impl/Documentation/DocstringConverter.cs

```diff
@@ -0,0 +1,460 @@
+﻿// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using Microsoft.Python.Core;
+
+namespace Microsoft.Python.LanguageServer.Documentation {
+    internal class DocstringConverter {
+        /// <summary>
+        /// Converts a docstring to a plaintext, human readable form. This will
+        /// first strip any common leading indention (like inspect.cleandoc),
+        /// then remove duplicate empty/whitespace lines.
+        /// </summary>
+        /// <param name="docstring">The docstring to convert, likely from the AST.</param>
+        /// <returns>The converted docstring, with Environment.NewLine line endings.</returns>
+        public static string ToPlaintext(string docstring) {
+            var lines = SplitDocstring(docstring);
+            var output = new List<string>();
+
+            foreach (var line in lines) {
```

Comment:
Can be full Linq if you want, condition in continue is Where.

Reply:
Not exactly, since that condition looks at the output to prevent empty lines from being duplicated. Best I could do is:

```csharp
var lines = SplitDocstring(docstring);
var output = lines.Where(
	(l, i) => !string.IsNullOrWhiteSpace(l) || !string.IsNullOrWhiteSpace(lines.ElementAtOrDefault(i - 1))
);
return string.Join(Environment.NewLine, output).TrimEnd();
```

Which I can do if you find that more readable.